administrator/components/com_installer/Model/ManageModel.php

```diff
@@ -324,7 +324,7 @@ protected function getListQuery()
 		$folder   = $this->getState('filter.folder');
 		$core     = $this->getState('filter.core');
```

Comment:
```suggestion
		$core     = $this->getState('filter.core', '');
```
Can it be done this way? If yes, then there is no to check null later.

Reply:
This should already be an empty string. Except for client filter because it defaults to `null` here https://github.com/joomla/joomla-cms/blob/4c3bd6b93be32938f6dae43a48e109af4be17f8e/administrator/components/com_installer/Model/ManageModel.php#L81.

Reply:
I tested filter values.
Just after a call from **backend form** I get:
$status: string '' (length=0)
$type: string '' (length=0)
$clientId: string '' (length=0)
$folder: string '' (length=0)
$core: string '' (length=0)
**and the query is listing all the extensions:**
string 'SELECT *,2*protected+(1-protected)*enabled AS status FROM #__extensions WHERE state = 0' (length=91)

After a **ws call** I get
$status: null
$type: null
$clientId: null
$folder: null
$core: null
**and the query returns no results:**
string 'SELECT *,2*protected+(1-protected)*enabled AS status FROM #__extensions WHERE state = 0 AND protected = 0 AND enabled = 0 AND client_id = 0 AND folder = ''' (length=159)

@SharkyKZ it seems that populateState isn't reached calling a ws, therefore all filters are null.

`Folder` filter is similar to `type`, so I can check just `if ($folder)`, but for status, client_id and core I need a fix like @Quy suggested:
`$this->getState('filter.status','');`
`$this->getState('filter.client_id','');`
`$this->getState('filter.core','');`

Reply:
@SharkyKZ, I'm struggling on what you suggested, populateState isn't reached and I guess that this could be a bug... I noticed that pagination is not working at all. The links for pagination of every ws (not only this one) return the same result (the first 20 records)... even changing url params values I get the same results. Perhaps solving this problem and invoking populateState will avoid the checks on filters, like you suggested...